### PR TITLE
park feature (Spyder)

### PR DESCRIPTION
**Pull request by JaskRendix**
PR implements park feature.

The first step is done. The area where the park is placed isn't accessible at the moment.

- adds new encounters **park** for wild monster -> to complete (only pairaigrin for testing purposes);
- adds **MainParkMenuState** (new state + file in states/combat);
- adds 2 new environments (**park** and **night_park**);
- adds new Tuxeball, [Park Tuxeball](https://wiki.tuxemon.org/Park_Tuxeball);
- adds new maps (**entrance** + **park** + **cave**);
- updates **lion_mountain** map to add the park entrance;
- updates **core_** by adding a bigger bridge (the same one, but bigger)

next steps:
- disabling wild monster attack;
- expanding the cave (?);
- formatting steps variable (it appears like 100.2524284) -> solution here: #2234 (event action: **format_variable**)

### CODE REVIEW

**Comment by Sanglorian** [comment]
This is a great addition! Excited to see it in the game, especially because we can use it to make available monsters that aren't incorporated into the main campaign.

**Comment by JaskRendix** [comment]
@Sanglorian this is the page of the park: https://wiki.tuxemon.org/Park

there are some missing dialogues and description

**Comment by Sanglorian** [comment]
> @Sanglorian this is the page of the park: https://wiki.tuxemon.org/Park
> 
> there are some missing dialogues and description

Thanks! I think I've filled them all in now. 

I called it Pepper Park as an homage to the free content comic Pepper & Carrot.

File: tuxemon/event/actions/random_encounter.py [hunk 1]
```diff
@@ -100,12 +107,22 @@ def start(self) -> None:
...
             # "queueing" it will mean it starts after the top of the stack
             # is popped (or replaced)
-            self.session.client.queue_state(
-                "CombatState",
-                players=(player, npc),
-                combat_type="monster",
-                graphics=env.battle_graphics,
-            )
+            if self.state == "ParkState":
+                if "menu_park_encountered" not in player.game_variables:
```
Comment (bitcraft) [inline tuxemon/event/actions/random_encounter.py]: this is fine for now, but its not going to be useful for mods/campaigns without mods, so at some point we will need to rethink this

**Review by xirsoi — DISMISSED** [review]

**Review by xirsoi — APPROVED** [review]

**Pull request merged**